pycqed/measurement/multi_qubit_module.py

```diff
@@ -2875,62 +3003,79 @@ def measure_pygsti(qubits, f_LO, pygsti_gateset=None,
+        qb.prepare(drive='timedomain')
 
-    channels = []
-    for qb in qubits:
-        channels += [qb.RO_acq_weight_function_I()]
-        if qb.ro_acq_weight_type() in ['SSB', 'DSB']:
-            if qb.RO_acq_weight_function_Q() is not None:
-                channels += [qb.RO_acq_weight_function_Q()]
-    df = det.UHFQC_integrated_average_detector(
-        UHFQC=UHFQC, AWG=pulsar, channels=channels,
-        integration_length=pulse_separation,
-        nr_averages=max([qb.RO_acq_averages() for qb in qubits]),
-        values_per_point=3,
-        values_per_point_suffex=['_probe', '_measure', '_ref_measure'])
-
-    qbr_names = [qbr.name for qbr in measured_qubits]
-    sf = awg_swf2.RO_dynamic_phase_swf(
-        qbp_name=pulsed_qubit.name,
-        qbr_names=qbr_names,
-        phases=phases,
-        operation_dict=operation_dict,
-        pulse_separation=pulse_separation,
-        init_state=init_state,
-        cal_points=cal_points,
-        upload=upload)
-    MC.set_sweep_function(sf)
-    MC.set_sweep_points(phases)
-    MC.set_detector_function(df)
-    exp_metadata = {'pulse_separation': pulse_separation,
-                    'cal_points': cal_points,
-                    'f_LO': f_LO}
+    if hard_sweep_params is None:
+        hard_sweep_params = {
+            'phase': {'values': np.tile(np.linspace(0, 2*np.pi, 6)*180/np.pi, 2),
+                      'unit': 'deg'}
+        }
+
+    cal_states = CalibrationPoints.guess_cal_states(cal_states,
+                                                    for_ef=False)
+    cp = CalibrationPoints.multi_qubit([qb.name for qb in qubits], cal_states,
+                                       n_per_state=n_cal_points_per_state)
+
+    if prep_params is None:
+        prep_params = measured_qubits[0].preparation_params()
+    operation_dict = get_operation_dict(qubits)
+    sequences, hard_sweep_points = \
+        mqs.ro_dynamic_phase_seq(
+            hard_sweep_dict=hard_sweep_params,
+            qbp_name=pulsed_qubit.name, qbr_names=[qb.name for qb in
+                                                   measured_qubits],
+            operation_dict=operation_dict,
+            pulse_separation=pulse_separation,
+            init_state = init_state, prep_params=prep_params,
+            cal_points=cp, upload=False)
+
+    hard_sweep_func = awg_swf.SegmentHardSweep(
+        sequence=sequences, upload=upload,
+        parameter_name=list(hard_sweep_params)[0],
+        unit=list(hard_sweep_params.values())[0]['unit'])
+    MC.set_sweep_function(hard_sweep_func)
+    MC.set_sweep_points(hard_sweep_points)
+
+    det_name = 'int_avg{}_det'.format('_classif' if classified else '')
+    det_func = get_multiplexed_readout_detector_functions(
+        measured_qubits, nr_averages=max(qb.acq_averages() for qb in measured_qubits)
+    )[det_name]
+    MC.set_detector_function(det_func)
+
+    if exp_metadata is None:
+        exp_metadata = {}
+
+    hard_sweep_params = {
```

Comment:
@Vintrix Do you understand what is going on here, why are the hard sweep parameters now in degrees, while before they were in rad. Also they are hardcoded here. Also now they are repeated and tiled while before only tiled. This function in its current state I would not want in the qudev_master. We could also just delete it for now and keep it only in Proj/XLD, if its hard to fix.

Reply:
Yes, let's do that for now.